contracts/eosio.system/src/voting.cpp

```diff
@@ -140,6 +173,13 @@ namespace eosiosystem {
       if( set_proposed_producers( producers ) >= 0 ) {
```

Comment:
We do not need to change the behavior of the host function as it currently exists. We could choose to make an optimization where, after Savanna, the host function checks if the policy has changed, and if it has not changed not bother adding a new policy to the header (in that case it would return the same version).

Whether we implement that change in the host function or not, the contract is not responsible for checking if the policy it is about to propose is different than the last proposed one.